### PR TITLE
refactored buttons and button styles now available for use

### PR DESCRIPTION
**Pull request by wrightmartin**

### CODE REVIEW

File: assets/stylesheets/responsive/custom.scss [hunk 1]
```diff
@@ -171,57 +171,94 @@ input.use-datepicker[type=text] {
...
+    &:hover,
+    &:active,
+    &:focus {
+        background-color: darken(#edece8, 7.5%);
+        border: 1px solid #ddd;
+    }
+}
+
+.button-large {
+  font-size: 2.2em;
```
Comment (garethrees) [inline assets/stylesheets/responsive/custom.scss]: Maybe a bit _too_ large! `1.5em` looked better to me.

File: assets/stylesheets/responsive/custom.scss [hunk 2]
```diff
@@ -171,57 +171,94 @@ input.use-datepicker[type=text] {
...
+    color: #fff;
+    background-color: $button-bg;
+    border: none;
+    &:hover,
+    &:active,
+    &:focus,
+    &:visited:hover,
+    &:visited:active,
+    &:visited:focus {
+        background-color: darken($button-bg, 10%);
```
Comment (garethrees) [inline assets/stylesheets/responsive/custom.scss]: Do you need to pass the `$button-bg` to `darken`? Is there a way to inherit that from `.button`, so in effect we just have something like `background-color: darken(10%)`?

Comment (garethrees) [inline assets/stylesheets/responsive/custom.scss]: Nope :( http://stackoverflow.com/questions/14920801/sass-manipulate-inherited-property

File: assets/stylesheets/responsive/custom.scss [hunk 3]
```diff
@@ -171,57 +171,94 @@ input.use-datepicker[type=text] {
...
 }
 
-@include lte-ie7{
+a.button,
+.button,
+.button:visited {
+    @extend .button-base;
+    color: #fff;
+    background-color: $button-bg;
+    border: none;
```
Comment (garethrees) [inline assets/stylesheets/responsive/custom.scss]: Should go in `.button-base`?

File: assets/stylesheets/responsive/custom.scss [hunk 4]
```diff
@@ -171,57 +171,94 @@ input.use-datepicker[type=text] {
...
+    @extend .button-base;
+    background-color: $color_light_grey;
+    color: $color_black;
+    font-weight: normal;
+    &:hover,
+    &:active,
+    &:focus {
+        color: #333;
+        background-color: darken($color_light_grey, 7.5%);
+        color: $color_black;
```
Comment (garethrees) [inline assets/stylesheets/responsive/custom.scss]: This has two `color:`

File: assets/stylesheets/responsive/custom.scss (hunk 2 above)
Comment (wrightmartin) [inline assets/stylesheets/responsive/custom.scss]: That would be so useful

**Comment by wrightmartin** [comment]
> I'm beginning to wonder about having "button" classes applied to markup.

We can use @extends as a preference, sure, but having .button as a utility class has been really useful on other sites. 

Personally, I find ```.button```, ```.button--secondary``` in the markup really handy in terms of self-documentation. It tells you really clearly, this is a primary action, this is a secondary action, etc

**Comment by garethrees** [comment]
> We can use @extends as a preference, sure, but having .button as a utility class has been really useful on other sites. 

Yeah I don't have a strong opinion on that really. More that the buttons are just UI utilities and that the markup doesn't really need to know about them.

> Personally, I find ```.button```, ```.button--secondary``` in the markup really handy in terms of self-documentation. It tells you really clearly, this is a primary action, this is a secondary action, etc

Yeah, I can see the appeal in that. Maybe `.action-bar__make-request--primary-action` and `.action-bar__follow-authority--secondary-action`?

File: assets/stylesheets/responsive/custom.scss [hunk 5]
```diff
@@ -171,57 +171,89 @@ input.use-datepicker[type=text] {
...
+  }
+  @if $disabled {
+    opacity: 0.333;
+    &:hover,
+    &:active,
+    &:focus {
+      background-color: inherit;
+      color: inherit;
+      cursor: default;
+    }
```
Comment (garethrees) [inline assets/stylesheets/responsive/custom.scss]: Disabled buttons shouldn't have the background colour transition

File: assets/stylesheets/responsive/custom.scss [hunk 6]
```diff
@@ -171,57 +171,89 @@ input.use-datepicker[type=text] {
...
+    &:hover,
+    &:active,
+    &:focus {
+      color: #333;
+      background-color:  desaturate(darken($color_sand, 15%), 5%);
+      color: $color_black;
+    }
+}
+
+@mixin button-tertiary($args...) {
```
Comment (garethrees) [inline assets/stylesheets/responsive/custom.scss]: A tertiary button is almost invisible:

screen shot 2015-08-07 at 14 13 44

Disabled teritary:

screen shot 2015-08-07 at 14 12 29

**Comment by garethrees** [comment]
Looks good on the whole.

- Need to make `.report-this-request` a button
- `button-tertiary` is too low-contrast
- Some uses of colour codes rather than colour variables
- Some 4-space instead of 2-space indents

**Comment by wrightmartin** [comment]
All done and ready to review again

**Pull request closed without merging**